From 9f5f72bcfb02ac28f905d5e845d6d84290e829a3 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:48 -0400
Subject: [fs] nfsd: make set of large acl return efbig, not resource

Message-id: <1403560314-32525-47-git-send-email-bfields@redhat.com>
Patchwork-id: 84608
O-Subject: [Patch RHEL7 046/172] nfsd4: make set of large acl return efbig, not resource
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

If a client attempts to set an excessively large ACL, return
NFS4ERR_FBIG instead of NFS4ERR_RESOURCE.  I'm not sure FBIG is correct,
but I'm positive RESOURCE is wrong (it isn't even a well-defined error
any more for NFS versions since 4.1).

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 798df3387971abf6071de77ca82b8e7775e74809
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 668bfe1..de7bc8f 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -294,7 +294,7 @@ nfsd4_decode_fattr(struct nfsd4_compoundargs *argp, u32 *bmval,
   READ32(nace);
 
   if (nace > NFS4_ACL_MAX)
-   return nfserr_resource;
+   return nfserr_fbig;
 
   *acl = nfs4_acl_new(nace);
   if (*acl == NULL)
-- 
1.7.1